client/src/reducers/documentsReducer.js

```diff
@@ -0,0 +1,13 @@
+export default function documentsReducer(state, action) {
+  const newState = Object.assign({}, state);
+  switch (action.type) {
+    case 'FETCH_DOCUMENTS_SUCCESS':
+      newState.count = action.payload.documents.length;
+      newState.documents = action.payload.documents;
+      break;
+
+    default:
+      break;
```

Comment:
You can have each case return newState after doing the object manipulation. Then have default also return newState without manipulations. Of course the break(s) would no longer be needed.